api/src/test/java/org/pmiops/workbench/billing/BillingProjectBufferServiceTest.java

```diff
@@ -479,13 +479,26 @@ public void assignBillingProject_locking() throws InterruptedException, Executio
   @Test
   public void cleanBillingBuffer_creating() {
     DbBillingProjectBufferEntry entry = new DbBillingProjectBufferEntry();
+    entry.setFireCloudProjectName("foo");
     entry.setStatusEnum(CREATING, this::getCurrentTimestamp);
     entry.setCreationTime(getCurrentTimestamp());
     billingProjectBufferEntryDao.save(entry);
 
+    doReturn(new BillingProjectStatus().creationStatus(CreationStatusEnum.CREATING))
+        .when(fireCloudService)
+        .getBillingProjectStatus(entry.getFireCloudProjectName());
+    billingProjectBufferService.syncBillingProjectStatus();
+
     CLOCK.setInstant(NOW.plus(61, ChronoUnit.MINUTES));
     billingProjectBufferService.cleanBillingBuffer();
 
+    // Time has elapsed but no status sync has occurred, should not transition.
+    assertThat(billingProjectBufferEntryDao.findOne(entry.getId()).getStatusEnum())
+        .isEqualTo(CREATING);
+
+    // Sync again, now @ +61 minutes - should transition.
+    billingProjectBufferService.syncBillingProjectStatus();
+    billingProjectBufferService.cleanBillingBuffer();
     assertThat(billingProjectBufferEntryDao.findOne(entry.getId()).getStatusEnum())
```

Comment:
Are you asserting that the first one you find is an error? Should they not all be in the error state?

Reply:
That's probably good enough (since you are familiar with the implementation), but I'd want to make sure it's operating on all and only the entries it's supposed to. It's easy to imagine a bug that happened only when more than 1 (or 5) entries were present.

Not a task for this PR though.

Reply:
That's the other test case that I added in response to your other comment: `cleanBillingBuffer_multipleCreates`. I'm not sure what more you're looking for in this test case.

Reply:
Oh, I'm sorry. I lost track of what was new.